src/internal_modules/roc_audio/latency_monitor.cpp

```diff
@@ -102,6 +104,12 @@ bool LatencyMonitor::reclock(const core::nanoseconds_t playback_timestamp) {
 }
 
 bool LatencyMonitor::pre_process_(const Frame& frame) {
+    if (fec_reader_) {
+        latency_metrics_.fec_block_duration =
+            packet_sample_spec_.stream_timestamp_2_ns(fec_reader_->max_block_duration());
+    } else {
+        latency_metrics_.fec_block_duration = 0;
+    }
```

Comment:
In LatencyMonitor::read, we have:

```
        compute_niq_latency_();
        query_link_meter_();

        if (!pre_process_(frame)) {
            alive_ = false;
        }
```

All metrics except fec_block_duration are filled before calling pre-process. So for consistency, I suggest to add a method like query_fec_reader() and also call it there.